subprojects/core/src/integTest/groovy/org/gradle/api/internal/tasks/TaskCacheabilityReasonIntegrationTest.groovy

```diff
@@ -16,131 +16,119 @@
-                disabledReasonCategory = TaskOutputCachingDisabledReasonCategory.BUILD_CACHE_DISABLED
-            }
-            task notcacheable(type: NotCacheable) {
-                cachingEnabled = false
-                disabledReason = "Task output caching is disabled"
-                disabledReasonCategory = TaskOutputCachingDisabledReasonCategory.BUILD_CACHE_DISABLED
-            }
+            task cacheable(type: Cacheable) {}
+            task notcacheable(type: NotCacheable) {}
         """
-        expect:
-        succeeds "cacheable", "notcacheable"
+        when:
+        run "cacheable"
+        then:
+        assertCachingDisabledFor BUILD_CACHE_DISABLED, "Build cache is disabled"
+
+        when:
+        run "notcacheable"
+        then:
+        assertCachingDisabledFor BUILD_CACHE_DISABLED, "Build cache is disabled"
     }
 
     def "cacheability for non-cacheable task is NOT_ENABLED_FOR_TASK"() {
         buildFile << """
-            task cacheable(type: NotCacheable) {
-                cachingEnabled = false
-                disabledReason = "Caching has not been enabled for the task"
-                disabledReasonCategory = TaskOutputCachingDisabledReasonCategory.NOT_ENABLED_FOR_TASK
-            }
+            task notcacheable(type: NotCacheable) {}
         """
-        expect:
-        withBuildCache().run "cacheable"
+        when:
+        withBuildCache().run "notcacheable"
+        then:
+        assertCachingDisabledFor NOT_ENABLED_FOR_TASK, "Caching has not been enabled for the task"
     }
 
     def "cacheability for a cacheable task is null"() {
         buildFile << """
-            task cacheable(type: Cacheable) {
-                cachingEnabled = true
-                disabledReason = null
-                disabledReasonCategory = null
-            }
+            task cacheable(type: Cacheable) {}
         """
-        expect:
+        when:
         withBuildCache().run "cacheable"
+        then:
+        assertCachingDisabledFor null, null
     }
 
     def "cacheability for a task with no outputs is NO_OUTPUTS_DECLARED"() {
         buildFile << """
             @CacheableTask
-            class NoOutputs extends BaseTask {
+            class NoOutputs extends DefaultTask {
                 @TaskAction
                 void generate() {}
             }
             
-            task cacheable(type: NoOutputs) {
-                cachingEnabled = false
-                disabledReason = "No outputs declared"
-                disabledReasonCategory = TaskOutputCachingDisabledReasonCategory.NO_OUTPUTS_DECLARED
-            }
+            task noOutputs(type: NoOutputs) {}
         """
-        expect:
-        withBuildCache().run "cacheable"
+        when:
+        withBuildCache().run "noOutputs"
+        then:
+        assertCachingDisabledFor NO_OUTPUTS_DECLARED, "No outputs declared"
     }
 
     def "cacheability for a task with no actions is UNKNOWN"() {
```

Comment:
Can we also have a test for cacheability of a task without actions which is not cacheable?

Reply:
Added.